catalyst/experiments/hydra_config.py

```diff
@@ -0,0 +1,552 @@
+                models, layerwise_params, no_bias_weight_decay, lr_scaling
+            )
+        elif isinstance(model_keys, str):
+            model_params = process_model_params(
+                models[model_keys],
+                layerwise_params,
+                no_bias_weight_decay,
+                lr_scaling,
+            )
+        elif isinstance(model_keys, (list, tuple, ListConfig)):
+            model_params = []
+            for model_key_el in model_keys:
+                model_params_el = process_model_params(
+                    models[model_key_el],
+                    layerwise_params,
+                    no_bias_weight_decay,
+                    lr_scaling,
+                )
+                model_params.extend(model_params_el)
+        else:
+            raise ValueError(
+                f"unknown type of model_params {type(model_keys)}"
+            )
+        # getting load-from-previous-stage flag
+        load_from_previous_stage = params.pop(
+            "load_from_previous_stage", False
+        )
+        # instantiate optimizer
+        optimizer: Optimizer = hydra.utils.instantiate(
+            params, params=model_params
+        )
+        # load from previous stage
+        if load_from_previous_stage and self.stages.index(stage) != 0:
+            checkpoint_path = f"{self.logdir}/checkpoints/best_full.pth"
+            checkpoint = load_checkpoint(checkpoint_path)
+            dict2load = optimizer
+            if name is not None:
+                dict2load = {name: optimizer}
+            unpack_checkpoint(checkpoint, optimizer=dict2load)
+            # move optimizer to device
+            device = get_device()
+            for param in model_params:
+                param = param["params"][0]
+                optimizer_state = optimizer.state[param]
+                for state_key, state_value in optimizer_state.items():
+                    optimizer_state[state_key] = any2device(
+                        state_value, device
+                    )
+            # update optimizer params
+            for key, value in params.items():
+                for optimizer_param_group in optimizer.param_groups:
+                    optimizer_param_group[key] = value
+
+        return optimizer
+
+    def get_optimizer(
+        self, stage: str, model: Dict[str, Model]
+    ) -> Dict[str, Optimizer]:
+        """
+        Returns the optimizers for a given stage.
+
+        Args:
+            stage (str):  stage name
+            model (Dict[str, Model]): dict of models
+
+        Returns:
+            Dict[str, Optimizer]: optimizer for selected stage
+
+        """
+        if "optimizers" not in self._config.stages[stage]:
+            return {}
+        optimizers_params: DictConfig = self._config.stages[stage].optimizers
+        optimizer = {
+            name: self._get_optimizer(stage, model, name, optimizer_params)
+            for name, optimizer_params in optimizers_params.items()
+        }
```

Comment:
is there any support for non-key-value optimizer?

Reply:
No. All are in key-value style (models, optimizers, schedulers, datasets, loaders, transforms, callbacks).
We do not need "key_val" flag in this case. Think this method is much more simpler and stable because of the unification.